src/client/Client.js

```diff
@@ -402,10 +408,25 @@ class Client extends BaseClient {
 
     const query = new URLSearchParams({
       client_id: this.application.id,
-      scope: 'bot',
     });
 
-    if (options.permissions) {
+    const { scopes } = options;
+    if (typeof scopes === 'undefined') {
+      throw new TypeError('INVITE_MISSING_SCOPES');
+    }
+    if (!Array.isArray(scopes)) {
+      throw new TypeError('INVALID_TYPE', 'scopes', 'Array of Invite Scopes', true);
+    }
+    if (!scopes.some(scope => ['bot', 'applications.commands'].includes(scope))) {
+      throw new TypeError('INVITE_MISSING_SCOPES');
+    }
+    const invalidScope = scopes.find(scope => !InviteScopes.includes(scope));
+    if (invalidScope) {
+      throw new TypeError('INVALID_ELEMENT', 'Array', 'scopes', invalidScope);
+    }
+    query.set('scope', scopes.join(' '));
+
+    if (options.permissions && options.scopes) {
```

Comment:
Isn't that redundant? Line 417 asserts that it's an array and no array is falsy.
```suggestion
    if (options.permissions) {
```

Reply:
Yeah this is a carry-over from when I thinking about having default scopes.

Should we only bother with permissions if the bot scope is included? It doesn't matter if they're in the URL, Discord just ignores them for applications.commands.

Reply:
Imo, we can just add it, don't have to check the presence of `bot` ourselves

CC @discordjs/the-big-3

Reply:
We don't have to, it could be that in a near future, Discord adds support for permissions in slash commands.